### PR TITLE
feat(atomic): added preprocessing for field values in atomic-result-multi-value-text

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-985

I made it possible for field values to be split and trimmed in `atomic-result-multi-value-text`.

### CODE REVIEW

File: packages/atomic/cypress/integration/result-list/result-components/result-multi-value-text.cypress.ts [hunk 1]
```diff
@@ -79,7 +87,7 @@ describe('Result MultiValueText Component', () => {
         .init();
     });
 
-    assertRemovesComponent(ResultMultiValueTextSelectors.shadow);
+    assertRemovesComponent(() => cy.get(resultMultiValueTextComponent));
```
Comment (btaillon) [inline packages/atomic/cypress/integration/result-list/result-components/result-multi-value-text.cypress.ts:90]: These selectors were entering the shadowDOM of elements, which means that Cypress asserted that these elements exist in order to access their shadowDOM before asserting that they don't.

File: packages/atomic/cypress/integration/result-list/result-components/result-multi-value-text.cypress.ts [hunk 2]
```diff
@@ -91,24 +99,29 @@ describe('Result MultiValueText Component', () => {
...
+        ResultListSelectors.firstResult().find(resultMultiValueTextComponent)
+      );
       assertConsoleError();
     });
 
-    describe('when the field value exists & is a string array', () => {
+    function testWithValidFieldValue(
+      getFieldValues: (valuesToTest: string[]) => string | string[],
+      delimiter?: string
+    ) {
```
Comment (btaillon) [inline packages/atomic/cypress/integration/result-list/result-components/result-multi-value-text.cypress.ts:124]: I'm repeating this whole series of test twice. Once with a string array, and once with a single string and a delimiter.

I want to ensure both are treated equally.

File: packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx [hunk 3]
```diff
@@ -50,6 +50,11 @@ export class AtomicResultMultiText {
    */
   @Prop() public maxValuesToDisplay = 3;
 
+  /**
+   * If the field isn't indexed as a multi value field, values are going to be split using this delimiter.
+   */
+  @Prop() public delimiter: string | null = null;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx:56]: Could the delimiter be purely CSS (::after & content)

File: packages/atomic/src/components.d.ts [hunk 4]
```diff
@@ -1691,6 +1695,10 @@ declare namespace LocalJSX {
         "imageSize"?: ResultDisplayImageSize;
     }
     interface AtomicResultMultiValueText {
+        /**
+          * If the field isn't indexed as a multi value field, values are going to be split using this delimiter.
```
Comment (dlutzCoveo) [inline packages/atomic/src/components.d.ts:1699]:
```suggestion
          * The delimiter used to separate values when the field isn't indexed as a multi value field.
```

File: packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx [hunk 5]
```diff
@@ -50,6 +50,11 @@ export class AtomicResultMultiText {
    */
   @Prop() public maxValuesToDisplay = 3;
 
+  /**
+   * If the field isn't indexed as a multi value field, values are going to be split using this delimiter.
```
Comment (dlutzCoveo) [inline packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx:54]:
```suggestion
   * The delimiter used to separate values when the field isn't indexed as a multi value field.
```

**Review by dlutzCoveo — APPROVED** [review]

File: packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx (hunk 3 above)
Comment (btaillon) [reply packages/atomic/src/components/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx:56]: It's not added to the multi value text, it's used to split it.

**Review by samisayegh — APPROVED** [review]

**Pull request merged**